addon/components/ember-tbody/component.js

```diff
@@ -306,6 +309,11 @@ export default Component.extend({
     this.collapseTree.set('rowMetaCache', this.rowMetaCache);
     this.collapseTree.set('rows', rows);
 
+    run.schedule('actions', () => {
+      // eslint-disable-next-line ember-best-practices/no-side-effect-cp
+      this.set('dataTestRowCount', this.get('wrappedRows.length')); // eslint-disable-line ember/no-side-effects
```

Comment:
if this passes the test then LGTM but obviously this super sucks explicitly adding this sideeffectyness. Thankfully it's pretty limited side effect but still 🤢

Reply:
Right, as noted this is expanding on an already-existing side effect which caused the problem. An alternative here would be to simply stop offering the `data-tests-row-count` attribute. If we drop 1.12 then this would be rewritten using `didReceiveAttrs` and the count could be set there without any CP side effects.